src/main/java/edu/unc/lib/boxc/migration/cdm/services/CdmIndexService.java

```diff
@@ -0,0 +1,227 @@
+import edu.unc.lib.boxc.migration.cdm.exceptions.MigrationException;
+import edu.unc.lib.boxc.migration.cdm.exceptions.StateAlreadyExistsException;
+import edu.unc.lib.boxc.migration.cdm.model.CdmFieldInfo;
+import edu.unc.lib.boxc.migration.cdm.model.MigrationProject;
+import edu.unc.lib.boxc.migration.cdm.util.ProjectPropertiesSerialization;
+
+/**
+ * Service for populating and querying the index of exported CDM records for a migration project
+ * @author bbpennel
+ */
+public class CdmIndexService {
+    private static final Logger log = getLogger(CdmIndexService.class);
+    public static final String TB_NAME = "cdm_records";
+
+    private MigrationProject project;
+    private CdmFieldService fieldService;
+
+    /**
+     * Indexes all exported CDM records for this project
+     * @throws IOException
+     */
+    public void indexAll() throws IOException {
+        assertCollectionExported();
+
+        CdmFieldInfo fieldInfo = fieldService.loadFieldsFromProject(project);
+        List<String> exportFields = fieldInfo.listExportFields();
+        String insertTemplate = makeInsertTemplate(exportFields);
+
+        SAXBuilder builder = SecureXMLFactory.createSAXBuilder();
+        Connection conn = null;
+        try (DirectoryStream<Path> dirStream = Files.newDirectoryStream(project.getExportPath(), "export*.xml")) {
+            conn = openDbConnection();
+
+            Iterator<Path> it = dirStream.iterator();
+            while (it.hasNext()) {
+                Path exportPath = it.next();
+                outputLogger.info("Indexing file: {}", exportPath.getFileName());
+                Document doc = builder.build(exportPath.toFile());
+                indexDocument(doc, conn, insertTemplate, exportFields);
+            }
+        } catch (IOException e) {
+            throw new MigrationException("Failed to read export files: " + e.getMessage());
+        } catch (SQLException e) {
+            throw new MigrationException("Failed to update database: " + e.getMessage());
+        } catch (JDOMException e) {
+            throw new MigrationException("Failed to parse export file: " + e.getMessage());
+        } finally {
+            closeDbConnection(conn);
+        }
+
+        project.getProjectProperties().setIndexedDate(Instant.now());
+        ProjectPropertiesSerialization.write(project);
+    }
+
+    private void assertCollectionExported() {
+        if (project.getProjectProperties().getExportedDate() == null) {
+            throw new InvalidProjectStateException("Must complete an export of the collection prior to indexing");
+        }
+    }
+
+    private String makeInsertTemplate(List<String> exportFields) {
+        return "insert into " + TB_NAME + " values ("
+                + exportFields.stream().map(f -> "?").collect(Collectors.joining(","))
+                + ")";
+    }
+
+    private void indexDocument(Document doc, Connection conn, String insertTemplate, List<String> exportFields)
```

Comment:
Just a general question. Is there only one table?

Reply:
yes, currently its a single table